torch/ao/quantization/fx/utils.py

```diff
@@ -945,3 +953,69 @@ def find_patterns(
+    If `qconfig` or `dtype_with_constraints.dtype` is None, or the dtypes do not match, return True.
+    """
+    def _activation_post_process_satisfies_dtype_config_constraints(
+            activation_post_process: Union[ObserverBase, FakeQuantize],
+            dtype_with_constraints: DTypeWithConstraints,
+            debug_string: str) -> bool:
+        app_quant_min = getattr(activation_post_process, "quant_min", None)
+        app_quant_max = getattr(activation_post_process, "quant_max", None)
+        # TODO: for now, just use the existing eps value as scale_min. In the future, we should
+        # resolve the differences between the two, either by renaming eps or some other way
+        app_scale_min = getattr(activation_post_process, "eps", None)
+        backend_quant_min = dtype_with_constraints.quant_min_lower_bound
+        backend_quant_max = dtype_with_constraints.quant_max_upper_bound
+        backend_scale_min = dtype_with_constraints.scale_min_lower_bound
+        # check quantization ranges
+        if backend_quant_min is not None and backend_quant_max is not None:
+            if app_quant_min is None or app_quant_max is None:
+                warnings.warn("QConfig %s must specify 'quant_min' and 'quant_max', ignoring %s" %
+                              (debug_string, qconfig))
+                return False
+            elif app_quant_min < backend_quant_min or app_quant_max > backend_quant_max:
+                warnings.warn(("QConfig %s quantization range must fall within the backend's:\n"
+                              "QConfig range = (%s, %s), BackendConfig range = (%s, %s), ignoring %s") %
+                              (debug_string, app_quant_min, app_quant_max,
+                              backend_quant_min, backend_quant_max, qconfig))
+                return False
+        # check scale min
+        if backend_scale_min is not None:
+            if app_scale_min is None:
+                warnings.warn("QConfig %s must specify 'eps', ignoring %s" % (debug_string, qconfig))
+                return False
+            elif app_scale_min < backend_scale_min:
+                warnings.warn(("QConfig %s eps (%s) must be greater than or equal to "
+                              "the backend's min scale value (%s), ignoring %s") %
+                              (debug_string, app_scale_min, backend_scale_min, qconfig))
+                return False
+        return True
+
+    if qconfig is None or dtype_with_constraints.dtype is None:
+        return True
+
+    activation_post_process_ctr = qconfig.activation if is_activation else qconfig.weight
+    debug_string = "activation" if is_activation else "weight"
+    satisfies_constraints = True
+    if activation_post_process_ctr is not None:
+        activation_post_process = activation_post_process_ctr()
+        assert isinstance(activation_post_process, (ObserverBase, FakeQuantize))
```

Comment:
we have a helper function here: https://github.com/pytorch/pytorch/blob/master/torch/ao/quantization/quantize.py#L47